src/backend/distributed/connection/shared_connection_stats.c

```diff
@@ -109,6 +109,18 @@ static uint32 SharedConnectionHashHash(const void *key, Size keysize);
 
 
 PG_FUNCTION_INFO_V1(citus_remote_connection_stats);
+PG_FUNCTION_INFO_V1(invalidate_inactive_shared_connections);
+
+/*
+ * invalidate_inactive_shared_connections invalidates inactive
+ * and not used shared connections by removing from the global hash.
+ */
+Datum
+invalidate_inactive_shared_connections(PG_FUNCTION_ARGS)
+{
+	RemoveInactiveNodesFromSharedConnections();
```

Comment:
I'd prefer if `citus_remote_connection_stats` would just omit inactive connections instead of introducing this UDF.

Reply:
So are you suggesting that we call `RemoveInactiveNodesFromSharedConnections` at the beginning of `citus_remote_connection_stats`?

Reply:
I guess we'll need such a UDF  anyway for  https://github.com/citusdata/citus/pull/3765 ?

Reply:
(we can do what Marco suggests here, and  add the UDF in  that PR if that makes more sense)

Reply:
Both of them are okay with me, how do we want to proceed?

Reply:
decouple the PRs by removing the function from here and cherry-picking it in #3765.